### PR TITLE
Improve MSVC cmake and update vcpkg instruction

### PR DESCRIPTION
**Pull request by Adriankhl**
This PR refactors the MSVC part of `src/CmakeLists.txt`, such that it separates the compiler flags into three parts cl and clang-cl, cl only, and clang-cl only. Some unnecessary flags are removed for clang-cl

Also, the PR adds the instruction to install `gmp` and `jsoncpp` via vcpkg in `README.md`. The main reason in doing so is that the bundled libraries don't compile out-of-the-box for `clang-cl`.

## To do

This PR is Ready for Review.

Maybe `ENABLE_SYSTEM_JSONCPP=ON` by default, it seems like the problem has already been fixed: https://github.com/minetest/minetest/issues/4306

## How to test
Compile with `clang-cl`

me personally use ninja and git bash, where `CC` and `CXX` are set to `clang-cl` in `~/.bashrc`, so the cmake command looks like `cmake .. -GNinja -DCMAKE_TOOLCHAIN_FILE=/d/packages/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_TARGET_TRIPLET=x64-windows -DCMAKE_RC_COMPILER_INIT=llvm-rc -DCMAKE_BUILD_TYPE=Release -DENABLE_GETTEXT=OFF -DENABLE_CURSES=OFF -DENABLE_SYSTEM_JSONCPP=ON`

### CODE REVIEW

**Review by sfan5 — APPROVED** [review]
I could probably test this if I boot up a VM and install MCSV but that'd take way too long.
This looks good and if it really breaks anything I'm sure someone will come complaining.

TL;DR :+1:

**Review by rubenwardy — APPROVED** [review]

**Comment by rubenwardy** [comment]
Not tested, but looks good
